saleor/giftcard/utils.py

```diff
@@ -53,16 +55,19 @@ def add_gift_card_code_to_checkout(
     checkout.save(update_fields=["last_change"])
 
 
-def remove_gift_card_code_from_checkout(checkout: Checkout, gift_card_code: str):
-    """Remove gift card data from checkout by code.
+def remove_gift_card_code_from_checkout_or_error(
+    checkout: Checkout, gift_card_code: str
+) -> None:
+    """Remove gift card data from checkout by code or raise an error."""
 
-    Return information whether promo code was removed.
-    """
     if gift_card := checkout.gift_cards.filter(code=gift_card_code).first():
         checkout.gift_cards.remove(gift_card)
         checkout.save(update_fields=["last_change"])
-        return True
-    return False
+    else:
+        raise ValidationError(
+            "Cannot remove a gift card not attached to this checkout.",
+            code=CheckoutErrorCode.GIFT_CARD_NOT_APPLICABLE.value,
```

Comment:
Same as with Voucher. Are you sure that the code is correct? We are not adding the GC, so it could be hard to understand why in case of removing the GC, we're returning NOT_APPLICABLE error code.